src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -165,113 +166,91 @@ public MergeFeedsJob(Auth0UserProfile owner, Set<FeedVersion> feedVersions, Stri
-            // regional and MTC merge strategies.
-            tablesToMerge.addAll(Arrays.asList(GtfsPlusTable.tables));
-        }
-        int numberOfTables = tablesToMerge.size();
-        // Loop over GTFS tables and merge each feed one table at a time.
-        for (int i = 0; i < numberOfTables; i++) {
-            Table table = tablesToMerge.get(i);
-            if (mergeType.equals(REGIONAL) && table.name.equals(Table.FEED_INFO.name)) {
-                // It does not make sense to include the feed_info table when performing a
-                // regional feed merge because this file is intended to contain data specific to
-                // a single agency feed.
-                // TODO: Perhaps future work can generate a special feed_info file for the merged
-                //  file.
-                LOG.warn("Skipping feed_info table for regional merge.");
-                continue;
+            // Determine which tables to merge (only merge GTFS+ tables for MTC extension).
+            final List<Table> tablesToMerge = getTablesToMerge();
+            int numberOfTables = tablesToMerge.size();
+            // Before initiating the merge process, get the merge strategy to use, which runs some pre-processing to
+            // check for id conflicts for certain tables (e.g., trips and calendars).
+            if (mergeType.equals(SERVICE_PERIOD)) {
+                mergeFeedsResult.mergeStrategy = getMergeStrategy();
             }
-            if (table.name.equals(Table.PATTERNS.name) || table.name.equals(Table.PATTERN_STOP.name)) {
-                LOG.warn("Skipping editor-only table {}.", table.name);
-                continue;
+            // Skip merging process altogether if the failing condition is met.
+            if (MergeStrategy.FAIL_DUE_TO_MATCHING_TRIP_IDS.equals(mergeFeedsResult.mergeStrategy)) {
+                failMergeJob("Feed merge failed because the trip_ids are identical in the future and active feeds. A new service requires unique trip_ids for merging.");
+                return;
             }
-            double percentComplete = Math.round((double) i / numberOfTables * 10000d) / 100d;
-            status.update("Merging " + table.name, percentComplete);
-            // Perform the merge.
-            LOG.info("Writing {} to merged feed", table.name);
-            int mergedLineNumber = constructMergedTable(table, feedsToMerge, out);
-            if (mergedLineNumber == 0) {
-                LOG.warn("Skipping {} table. No entries found in zip files.", table.name);
-            } else if (mergedLineNumber == -1) {
-                LOG.error("Merge {} table failed!", table.name);
+            // Loop over GTFS tables and merge each feed one table at a time.
+            for (int i = 0; i < numberOfTables; i++) {
+                Table table = tablesToMerge.get(i);
+                if (shouldSkipTable(table.name)) continue;
```

Comment:
There is now a `shouldSkipTable` method and also within the MergeLineContext there is the `skipFile` field. Perhaps this logic can be consolidated somehow.

Reply:
This one will probably not be addressed, one of the skipping logic just looks at table names while the other does examine some of the table content in one particular feed before deciding whether to skip (that same table in the other feed to merge could still be processed).